src/function/table/read_csv.cpp

```diff
@@ -326,6 +326,25 @@ static unique_ptr<FunctionData> CSVReaderDeserialize(Deserializer &deserializer,
 	return std::move(result);
 }
 
+bool PushdownTypeToCSVScanner(LogicalGet &logical_get, const vector<LogicalType> &target_types,
+                              const vector<unique_ptr<Expression>> &expressions) {
+	auto &csv_bind = logical_get.bind_data->Cast<ReadCSVData>();
+	// We loop in the expressions and bail out if there is anything weird (i.e., not a bound column ref)
+	vector<LogicalType> pushdown_types = csv_bind.csv_types;
+	for (idx_t i = 0; i < expressions.size(); i++) {
+		if (expressions[i]->type == ExpressionType::BOUND_COLUMN_REF) {
+			auto &col_ref = expressions[i]->Cast<BoundColumnRefExpression>();
+			pushdown_types[logical_get.column_ids[col_ref.binding.column_index]] = target_types[i];
```

Comment:
I think we should have this logic where we actually call the `pushdown_type` method - all that is relevant for the CSV reader is the bind data and the target types. I suggest the following signature here:

```cpp
typedef void (* table_function_type_pushdown_t)(ClientContext &context, optional_ptr<FunctionData> bind_data, const unordered_map<idx_t, LogicalType> &new_column_types);
```

Then the assumption is that if a `type_pushdown` method is defined, this method can always be applied to override column types in the bind (i.e. it always succeeds). Extraction of the new types per column can be done generically - this way we can also extend it in the future if required.

Reply:
Why are we passing on the client context and not the LogicalGet?

Reply:
We should not be passing the `LogicalGet` as that's a higher level abstraction - table functions should keep their state in the bind data and we should manage the `LogicalGet` outside of these calls.

The `ClientContext` is more about future-proofing, since that allows us to access e.g. settings

Reply:
Alright, thanks for the clarification! :-)

Reply:
I guess the unordered_map instead of vector<LogicalType> is also future-proofing?